lib/src/components/_stacks-progress-bars.less

```diff
@@ -0,0 +1,182 @@
+//  visit https://stackoverflow.design/
+//
+//  TABLE OF CONTENTS
+//  • VARIABLE
+//  • BASE STYLE
+//  • VARIATION PROGRESS BACKGROUNDS
+//  • BADGES
+//
+//  ============================================================================
+//  $   VARIABLES
+//  ----------------------------------------------------------------------------
+@progressbar-zindex:       0;
+@progressbar-fill-bgc:     @black-100;
+
+
+//  ============================================================================
+//  $   BASE STYLE
+//  ----------------------------------------------------------------------------
+.s-progress {
+    position: relative;
+    width: 100%;
+    min-height: @su4;
+}
+
+//  ============================================================================
+//  $   CHILD STYLES
+//  ----------------------------------------------------------------------------
+.s-progress--label {
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    border-radius: @br-md;
+    border: 1px solid transparent;
+    font-size: @fs-caption;
+    color: @fc-medium;
+    z-index: @progressbar-zindex + 2;
+}
+
+.s-progress--fill,
+.s-progress--bar {
+    position: absolute;
+    height: 100%;
+}
+
+//  $$   FILL
+//       The fill is the background element. We do this so we can properly use
+//       background gradients.
+//  ----------------------------------------------------------------------------
+.s-progress--fill {
+    z-index: @progressbar-zindex;
+    left: 0;
+    right: 0;
+    width: 100%;
+    background-repeat: no-repeat;
+    background-position: 0 0;
+    background-size: cover;
+    background-image: linear-gradient(to right, #5ED286 0, #23D59B 100%);
+    border-radius: @br-sm;
+}
+
+//  $$   BAR
+//       A gray mask that reveals the fill.
+//  ----------------------------------------------------------------------------
+.s-progress--bar {
+    //   Pseudo elements to make it look like the background gradient is the bar.
+    &:before,
+    &:after {
+        content: "";
+        position: absolute;
+        left: -(@su2);
+        border: 2px solid transparent;
+    }
+    &:before {
+        top: 0;
+        border-top-color: @progressbar-fill-bgc;
+    }
+    &:after {
+        bottom: 0;
+        border-bottom-color: @progressbar-fill-bgc;
+    }
+
+    //   Hide the psuedo elements though when the bar is at 0% or 100%
+    &[aria-valuenow="0"],
+    &[aria-valuenow="100"] {
+      &:before,
+      &:after {
+        opacity: 0;
+      }
+    }
+
+    z-index: @progressbar-zindex + 1;
+    right: 0;
+    border-radius: @br-sm;
+    background-color: @progressbar-fill-bgc;
+}
+
+
+//  ===========================================================================
+//  $   VARIATION PROGRESS BACKGROUNDS
+//  ---------------------------------------------------------------------------
+.s-progress__brand .s-progress--fill {
+    background-image: linear-gradient(to right, #F58907 0%, #F16720 100%);
```

Comment:
Should these be based on color variables?

Reply:
These are specific HEX values that either pre-existed (and don't exist as color variables currently), or were identified within the Visual Guide.